src/main/java/frc/robot/subsystems/Chassis.java

```diff
@@ -33,40 +39,116 @@
      * double cLR = 0.1; This was ctrl c ctrl v'd from 2020. I don't now what
      * these values mean so I don't want to delete them.
      */
-    private CANSparkMax initMotor(CANSparkMax sparky, int id){
+
+    private CANSparkMax initMotor(CANSparkMax sparky, int id, RelativeEncoder r) {
         sparky = new CANSparkMax(id, MotorType.kBrushless);
         SparkMaxPIDController pid = sparky.getPIDController();
         sparky.restoreFactoryDefaults();
+        sparky.setCANTimeout(TIMEOUT);
         final double kP= 6e-5;
         final double kI = 0;
         final double kD = 0;
         final double kIz = 0;
         final double kFF = 0.000015;
         final double kMaxOutput = 1;
         final double kMinOutput = -1;
-        final double maxRPM = 5700;
         pid.setP(kP);
         pid.setI(kI);
         pid.setD(kD);
         pid.setIZone(kIz);
         pid.setFF(kFF);
         pid.setOutputRange(kMinOutput, kMaxOutput);
+        sparky.setCANTimeout(TIMEOUT);
+        r = sparky.getEncoder();
```

Comment:
`r` is passed in, then set from the sparkmax instead, then discarded? Can you help me understand what you're trying to do?